### ISSUE TITLE
Thread pools and connectors

### ISSUE THREAD

**Issue by thomaslee** [eclipse/jetty.project#586]
Not entirely sure if this is a bug or a misunderstanding of expected behavior -- brought to my attention by @ajermakovics in https://github.com/dropwizard/dropwizard/issues/1477

When using multiple connectors, `Server.doStart()` seems to try and validate that the "primary" thread pool has enough threads for _all_ connectors -- even if one or more of those connectors have their own distinct thread pools.

The following test case demonstrates the issue: 1+1+1 threads required for connector 1, 1+1+1 required for connector 2, each connector gets its own thread pool (connector 1 inherits the server thread pool, though I don't think that's entirely relevant here), yet the server is unable to start because `doStart` wants the server's thread pool to have at least 6 threads available to it. Ignore lifecycle management issues for simplicity's sake, I don't believe they're relevant:

``` java
import org.eclipse.jetty.io.ArrayByteBufferPool;
import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.server.ServerConnector;
import org.eclipse.jetty.util.thread.QueuedThreadPool;
import org.junit.Test;

public class TrivialJettyTest {
    @Test
    public void canJettyWithMultipleConnectors() throws Exception {
        final QueuedThreadPool threadPool1 = new QueuedThreadPool(3, 3);
        final Server server = new Server(threadPool1);
        server.addConnector(new ServerConnector(server, null, null, new ArrayByteBufferPool(), 1, 1));

        final QueuedThreadPool threadPool2 = new QueuedThreadPool(3, 3);
        server.addConnector(new ServerConnector(server, threadPool2, null, new ArrayByteBufferPool(), 1, 1));

        server.start();
        //
        // nothing below this line will execute because start() throws an ISE.
        //
        try {
        }
        finally {
            server.stop();
        }
    }
}
```

Result is an IllegalStateException:

```
01:25:39.347 [main] INFO  org.eclipse.jetty.util.log - Logging initialized @1109ms
01:25:39.409 [main] INFO  org.eclipse.jetty.server.Server - jetty-9.3.9.v20160517

java.lang.IllegalStateException: Insufficient threads: max=3 < needed(acceptors=2 + selectors=2 + request=1)
    at org.eclipse.jetty.server.Server.doStart(Server.java:374)
    at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
    at io.dropwizard.testing.TrivialJettyTest.canJettyWithMultipleConnectors(TrivialJettyTest.java:19)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
    at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
    at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
    at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
    at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
    at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
    at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
    at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
    at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
    at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
    at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
    at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
    at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
    at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:119)
    at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:42)
    at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:234)
    at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:74)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

**Comment by sbordet** [comment]
Fixed by #1010.

**Comment by gregw** [comment]
Unit test created for this issue is regularly failing.  Ignoring it for now!
```
java.lang.IllegalStateException: Insufficient threads: max=3 < needed(acceptors=1 + selectors=2 + request=1)
	at org.eclipse.jetty.server.Server.doStart(Server.java:414)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:68)
	at org.eclipse.jetty.server.InsufficientThreadsDetectionTest.testCaseForMultipleConnectors(InsufficientThreadsDetectionTest.java:83)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

**Comment by sbordet** [comment]
@gregw you did not pull the latest code, the test failure was already fixed by 34d0aa3ab83923f234908f335cdcabfc75b59ba9.
